README.md

```diff
@@ -2,23 +2,163 @@
+Download the designated appropriate file for your device from the [latest release](https://github.com/redolution/iplboot/releases/latest).
+
+Prepare your SD card by copying DOLs onto the SD card and renaming them according the table above.
+
+### PicoBoot
+
+You can download ready-to-go firmware featuring IPLBoot directly from PicoBoot. Just follow the [installation guide](https://github.com/webhdx/PicoBoot#-installation-guide). This is likely what you want.
+
+Otherwise, use `iplboot.dol` and follow the PicoBoot [compiling firmware](https://github.com/webhdx/PicoBoot#compiling-firmware) instructions to flash your Raspberry Pico.
+
+### Qoob Pro
+
+Use `iplboot.gcb` to flash your Qoob Pro as a BIOS.
+
+### Qoob SX
+
+Qoob SX is not currently supported.
+
+### Viper
+
+Use `iplboot.vgc` to flash your Viper as a BIOS.
+
+### GameCube Memory Card
+
+Use `iplboot_xz.gci` and a GameCube memory card manager such as [GCMM](https://github.com/suloku/gcmm) to copy to your memory card. It will be saved as `boot.dol` and can be used in conjunction with the various [save game exploits](https://www.gc-forever.com/wiki/index.php?title=Booting_homebrew#Game_Save_Exploits).
+
+
+## Troubleshooting
+
+IPLBoot displays useful diagnostic messages as it attempts to load the selected DOL. But it's so fast you may not have time to read or even see them. If you hold the down direction on the D-Pad, the messages will remain on screen until you let go.
+
+
+## Compiling
+
+You should only need to compile if you want to modify to the IPLBoot source code. Otherwise, you can just download needed files from the [latest release](https://github.com/redolution/iplboot/releases/latest).
+
+1. Ensure you have the latest version of [devkitPro](https://devkitpro.org/wiki/Getting_Started) installed.
+2. Ensure you have the latest version of [libogc2](https://github.com/extremscorner/libogc2) installed (see [instructions](#installing-libogc2) below).
+3. Ensure your devkitPro environment variables are set: `DEVKITPRO` and `DEVKITPPC`
+
+IPLBoot also acts as a server for @emukidid's [usb-load](https://github.com/emukidid/gc-usb-load), should you want to use it for development purposes.
+
+### PicoBoot
+
+1. Run `make dol`.
+2. Follow usage instructions above using the newly created `build/iplboot.dol`.
+
+### Qoob Pro
+
+1. Obtain the gc-pal-10 IPL ROM (MD5: `0cdda509e2da83c85bfe423dd87346cc`) and copy it to the root of this project as `ipl.rom`. This is currently the only IPL that has been verified to work.
+2. Run `make qoobpro`.
+3. Follow usage instructions above using the newly created `build/iplboot.gcb`.
+
+### Qoob SX
+
+NOTE: Qoob SX is currently nonfunctional as the resulting BIOS is too large.
+
+1. Ensure dolxz is installed (see [instructions](#installing-dolxz) below).
+2. Ensure doltool is installed (see [instructions](#installing-doltool) below).
+3. Obtain `qoob_sx_13c_upgrade.elf` and copy it to the root of this project. (TODO: MD5 and more details about the file.)
```

Comment:
[Qoob (1.3c).rar](https://www.gc-forever.com/forums/download/file.php?id=109)
https://www.gc-forever.com/forums/viewtopic.php?f=36&t=23

Reply:
Thanks! How about?
```suggestion
3. Download [Qoob BIOS v1.3c](https://www.gc-forever.com/forums/download/file.php?id=109) and extract `qoob_sx_13c_upgrade.elf` (MD5: `2292630e7604bc2ae8bfa4d3e4ba5941`) to the root of this project.
```

Reply:
Looks good, but again instead of linking directly to the file, link to the page.

Reply:
Sure thing. Curious though, why not link directly to appropriate files?

Reply:
Well, in this particular case, the file won't get updated, but the page still provides some context for the file.